src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -101,15 +101,17 @@ private void RemoveFromConnectionIfDone()
             }
         }
 
-        public async ValueTask DisposeAsync()
+        public ValueTask DisposeAsync()
         {
             if (!_disposed)
             {
                 _disposed = true;
                 AbortStream();
-                await _stream.DisposeAsync().ConfigureAwait(false);
+                _connection.QueueStreamForDisposal(_stream);
```

Comment:
AFAIU, this should be in sync `Dispose`, the async `DisposeAsync` doen't have an issue with sync-over-async and is not even called in the scenario we're testing here.